### PR TITLE
Added GS and coloring examples.

### PR DESCRIPTION
**Pull request by brian-kelley**
Added optional argument to kk_generate_diagonally_dominant_sparse_matrix,
to control the level of diagonal dominance (scaling factor on absolute sum of off-diagonals).
This can be turned down to get a slightly more difficult linear system, that takes more than 2
GS iterations to reach small tolerance.

The coloring example creates a graph from 9-pt stencil on a rectangular 2D grid. It runs D1 and D2 coloring, and prints out the colors arranged in the grid. The GS example creates a diagonally dominant matrix and uses GS as an iterative solver, printing out the relative residual norm at each step.

Example outputs:
```
Generated a matrix with 10000 rows/cols, and 200000 entries.
Iteration 1 scaled residual norm: 0.186896
Iteration 2 scaled residual norm: 0.0282584
Iteration 3 scaled residual norm: 0.00360863
Iteration 4 scaled residual norm: 0.000432937
Iteration 5 scaled residual norm: 5.00684e-05
Iteration 6 scaled residual norm: 5.79139e-06
Iteration 7 scaled residual norm: 6.75833e-07
SUCCESS: converged in 7 iterations.
```
```
9-pt stencil: Distance-1 Colors (used 6):
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 4 3 4 3
 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1
 3 4 3 4 3 4 3 4 3 4 3 5 4 6 5
 1 2 1 2 1 2 1 2 1 2 1 2 1 3 2
 3 4 3 4 3 4 3 4 3 5 4 5 4 5 4
 1 2 1 2 1 2 1 2 1 6 2 3 2 3 2
 3 4 3 4 3 5 4 5 4 5 4 5 4 5 4
 1 2 1 2 1 2 6 3 2 3 2 3 2 3 2
 3 5 4 5 6 5 4 5 4 5 4 5 4 5 4
 1 2 1 3 2 3 2 3 2 3 2 3 2 3 2

9-pt stencil: Distance-2 Colors (used 15):
  1  2  3  1  2  3  1  2  3  1  2  3  1  4  3
  4  5  6  4  5  6  4  5  6  4  5  6  7  5  2
  7  8  9  7  8  9  7  8  9  7  8  9 10  8  6
  1  2  3  1  2  3  1  2  3  1  2  3  1  4  3
  4  5  6  4  5  6  4  5  6  4  5  6  7  5  9
  7  8  9  7  8  9  7  8  9  7  8  9 10  8  6
  1  2  3  1  2  3  1  2  3  1  2  3  1  4  2
  4  5  6  4  5  6  4  5  6  4  5  6  7  5  3
  7  8  9  7  8  9  7  8  9  7  8  9 10  8  6
  1  3 10  1  2  3  1  2  3  1  2  3  1  4  9
  4  5  6  4  5  6  4  5  6  4  5  6  7  5  2
  2  7  8  9  7  8  9  7  8  9 10  8 11 10  3
  1  3 10  1  2  3  1  2  3  1  2  3  1  4  6
  4  5  6  4  5  6  4  5  6  4  5  6  7  5  8
  2  7  8  9  7  8  9  7  8  9 10  8  9 10  3
  1  3 10  1  2  3  1  2  3  1  2  3  1  4  2
  4  5  6  4  5  6  4  5  6  4  5  6  7  5  6
  2  7  8  9  7  8  9  7  8  9 10 11 12 10  3
  1  3 10  1  2  3  1  2  3  1  2  3  1  4  8
  5  6 11  5  4  6  5  4  6 12  4  8  9 13  2
  2  4  7  8  9  7  8  9  7 10 11  5  6  7  5
  1  9 10  1  3 10  1  2 13  1 14 15  1 10  3
  3  5  6 11  4  5  6  3  4  6  3  8  2  4  8
  2  4  7  2  8  7  9  8  5 10  7  5  6  7  5
  1  8  9  1  3 10  1  2 11  1 12  9  1  3  9
```

### CODE REVIEW

**Comment by brian-kelley** [comment]
kokkos-dev2 (double & complex)
#######################################################
PASSED TESTS
#######################################################
clang-8.0-Cuda_OpenMP-release build_time=670 run_time=166
clang-8.0-Pthread_Serial-release build_time=220 run_time=128
clang-9.0.0-Pthread-release build_time=133 run_time=58
clang-9.0.0-Serial-release build_time=139 run_time=64
cuda-10.1-Cuda_OpenMP-release build_time=853 run_time=162
cuda-9.2-Cuda_Serial-release build_time=833 run_time=184
gcc-4.8.4-OpenMP-release build_time=116 run_time=55
gcc-7.3.0-OpenMP-release build_time=153 run_time=61
gcc-7.3.0-Pthread-release build_time=125 run_time=59
gcc-8.3.0-Serial-release build_time=146 run_time=59
gcc-9.1-OpenMP-release build_time=179 run_time=60
gcc-9.1-Serial-release build_time=166 run_time=63
intel-17.0.1-Serial-release build_time=261 run_time=63
intel-18.0.5-OpenMP-release build_time=364 run_time=55
intel-19.0.5-Pthread-release build_time=472 run_time=52

File: src/common/KokkosKernels_IOUtils.hpp [hunk 1]
```diff
@@ -169,7 +169,8 @@ void kk_diagonally_dominant_sparseMatrix_generate(
     OrdinalType bandwidth,
     ScalarType* &values,
     SizeType* &rowPtr,
-    OrdinalType* &colInd)
+    OrdinalType* &colInd,
+    ScalarType diagDominance = 10 * Kokkos::ArithTraits<ScalarType>::one())
```
Comment (ndellingwood) [inline src/common/KokkosKernels_IOUtils.hpp:173]: @seheracer Brian's change here may be helpful for your detective work with float type in spgemm_jacobi (I think you had observations about the scalar multiplier's value on diagonally dominant matrices having impact on convergence)

**Comment by brian-kelley** [comment]
Thanks @ndellingwood . I'm adapting these into actual Begin/Solution exercises in kokkos-tutorials now.

File: src/common/KokkosKernels_IOUtils.hpp (hunk 1 above)
Comment (seheracer) [reply src/common/KokkosKernels_IOUtils.hpp:173]: Thanks, @ndellingwood for letting me know. This is a useful feature @brian-kelley, thanks!

**Pull request merged**